language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -652,7 +652,65 @@ impl CompiledPackage {
         Ok((compiled_package, CompilationCachingStatus::Recompiled))
     }
 
+    // We take the (restrictive) view that all filesystems are case insensitive to maximize
+    // portability of packages.
+    fn check_filepaths_ok(&self) -> Result<()> {
+        // A mapping of (lowercase_name => (real_name, [info_for_each_occurence])
+        let mut insensitive_mapping = BTreeMap::new();
+        for compiled_unit in &self.compiled_units {
+            let is_module = matches!(&compiled_unit.unit, CompiledUnit::Module(_));
+            let name = match &compiled_unit.unit {
+                CompiledUnit::Script(named) => named.name.as_str(),
+                CompiledUnit::Module(named) => named.name.as_str(),
+            };
+            let entry = insensitive_mapping
+                .entry(name.to_lowercase())
+                .or_insert_with(BTreeMap::new);
+            let insert = entry.entry(name).or_insert_with(Vec::new);
```

Comment:
This is a weird name. I don't understand what is going on here because it is called `insert` I would expect this to be some plural noun.